src/ConfigHelper.js

```diff
@@ -113,16 +113,22 @@ function configureProps() {
 }
 
 export function addWhitelistedNativeProps(props) {
+  const oldSize = Object.keys(NATIVE_THREAD_PROPS_WHITELIST).length;
   NATIVE_THREAD_PROPS_WHITELIST = {
     ...NATIVE_THREAD_PROPS_WHITELIST,
     ...props,
   };
-  configureProps();
+  if (oldSize !== Object.keys(NATIVE_THREAD_PROPS_WHITELIST).length) {
+    configureProps();
+  }
 }
 
 export function addWhitelistedUIProps(props) {
+  const oldSize = Object.keys(UI_THREAD_PROPS_WHITELIST).length;
   UI_THREAD_PROPS_WHITELIST = { ...UI_THREAD_PROPS_WHITELIST, ...props };
-  configureProps();
+  if (oldSize !== Object.keys(UI_THREAD_PROPS_WHITELIST).length) {
+    configureProps();
+  }
```

Comment:
I'm assuming this is an optimization to not run the update when `UI_THREAD_PROPS_WHITELIST ` is unchanged?

What if we change `propertyName: true` to `propertyName: false`? It won't change the length.

Reply:
Yes, I know it will not. Look at the `configureProps` function - only keys are considered.

Reply:
You're right. I checked native implementation and we use only keys. I think we should replace an object with a set in some follow-up PR, so it's clearer what it does.